packages/utils/telemetry-utils/src/utils.ts

```diff
@@ -68,32 +68,48 @@ export function createSampledLogger(
 			// 1. If isSamplingDisabled is true, then this means events should be unsampled. Therefore we send the event without any checks.
 			// 2. If isSamplingDisabled is false, then event should be sampled using the event sampler, if the sampler is not defined just send all events, other use the eventSampler.sample() method.
 			// 3. If skipLoggingWhenSamplingIsDisabled is true, then no event is sent.
-			if (isSamplingDisabled || eventSampler === undefined || eventSampler.sample()) {
-				if (isSamplingDisabled && skipLoggingWhenSamplingIsDisabled) {
+			if (
+				isSamplingDisabled ||
+				eventSampler === undefined ||
+				(eventSampler.sample() ?? false)
```

Comment:
Instead of this one I think we can make `IEventSampler.sample()` return just `boolean` instead of `boolean | undefined`. I don't see why it would ever need to return undefined. It's an internal interface, and all the implementations already return a boolean, so it's a safe change to make.

Reply:
I'll update it.